RFXCOM NPE when not even connected

While the bridge is not connected, it is not even plugged in... it's still trying to send out the message and fails because there is OutputStream available that exception not caught because it's an NPE so finally we get a real error in the log.

```
2017-05-25 19:36:27.743 [ERROR] [.c.thing.internal.ThingManager:363  ] - Exception occured while calling handler: java.lang.NullPointerExceptionjava.util.concurrent.ExecutionException: java.lang.NullPointerException
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:206)
	at org.eclipse.smarthome.core.common.SafeMethodCaller.callAsynchronous(SafeMethodCaller.java:186)
	at org.eclipse.smarthome.core.common.SafeMethodCaller.call(SafeMethodCaller.java:79)
	at org.eclipse.smarthome.core.common.SafeMethodCaller.call(SafeMethodCaller.java:63)
	at org.eclipse.smarthome.core.thing.internal.ThingManager.receiveCommand(ThingManager.java:352)
	at org.eclipse.smarthome.core.items.events.AbstractItemEventSubscriber.receive(AbstractItemEventSubscriber.java:46)
	at org.eclipse.smarthome.core.internal.events.OSGiEventManager$1.call(OSGiEventManager.java:192)
	at org.eclipse.smarthome.core.internal.events.OSGiEventManager$1.call(OSGiEventManager.java:1)
	at org.eclipse.smarthome.core.common.SafeMethodCaller$CallableWrapper.call(SafeMethodCaller.java:177)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
Caused by: java.lang.NullPointerException: null
	at org.openhab.binding.rfxcom.internal.connector.RFXComSerialConnector.sendMessage(RFXComSerialConnector.java:126)
	at org.openhab.binding.rfxcom.handler.RFXComBridgeHandler.sendMessage(RFXComBridgeHandler.java:215)
	at org.openhab.binding.rfxcom.handler.RFXComHandler.handleCommand(RFXComHandler.java:83)
	at org.eclipse.smarthome.core.thing.internal.ThingManager$3.call(ThingManager.java:355)
	at org.eclipse.smarthome.core.thing.internal.ThingManager$3.call(ThingManager.java:1)
	... 5 common frames omitted
```

If the serial connector sees that it has no correct OutputStream it should maybe throw an IOException which is properly handled in the bridge handler and it will keep/mark the item offline with a correct message.

--- a/addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComBridgeHandler.java
+++ b/addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/handler/RFXComBridgeHandler.java
@@ -11,8 +11,8 @@ package org.openhab.binding.rfxcom.handler;
 import java.io.IOException;
 import java.util.HashSet;
 import java.util.List;
-import java.util.Set;
 import java.util.Queue;
+import java.util.Set;
 import java.util.concurrent.CopyOnWriteArrayList;
 import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.ScheduledFuture;
@@ -41,11 +41,9 @@ import org.openhab.binding.rfxcom.internal.messages.RFXComInterfaceControlMessag
 import org.openhab.binding.rfxcom.internal.messages.RFXComInterfaceMessage;
 import org.openhab.binding.rfxcom.internal.messages.RFXComInterfaceMessage.Commands;
 import org.openhab.binding.rfxcom.internal.messages.RFXComInterfaceMessage.SubType;
-import org.openhab.binding.rfxcom.internal.messages.RFXComInterfaceMessage.TransceiverType;
 import org.openhab.binding.rfxcom.internal.messages.RFXComMessage;
 import org.openhab.binding.rfxcom.internal.messages.RFXComMessageFactory;
 import org.openhab.binding.rfxcom.internal.messages.RFXComTransmitterMessage;
-import org.openhab.binding.rfxcom.internal.messages.RFXComTransmitterMessage.Response;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -109,7 +107,6 @@ public class RFXComBridgeHandler extends BaseBridgeHandler {
     }
 
     private TransmitQueue transmitQueue = new TransmitQueue();
-
 
     public RFXComBridgeHandler(Bridge br) {
         super(br);
@@ -224,7 +221,7 @@ public class RFXComBridgeHandler extends BaseBridgeHandler {
             transmitQueue.enqueue(baseMsg);
         } catch (IOException e) {
             logger.error("I/O Error", e);
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
         }
     }
 
@@ -259,12 +256,14 @@ public class RFXComBridgeHandler extends BaseBridgeHandler {
                                         logger.warn("Failed to parse setMode data", ee);
                                     }
                                 } else {
-                                    RFXComInterfaceControlMessage modeMsg = new RFXComInterfaceControlMessage(msg.transceiverType, configuration);
+                                    RFXComInterfaceControlMessage modeMsg = new RFXComInterfaceControlMessage(
+                                            msg.transceiverType, configuration);
                                     setMode = modeMsg.decodeMessage();
                                 }
 
                                 if (setMode != null) {
-                                    logger.debug("Setting RFXCOM mode using: {}", DatatypeConverter.printHexBinary(setMode));
+                                    logger.debug("Setting RFXCOM mode using: {}",
+                                            DatatypeConverter.printHexBinary(setMode));
                                     connector.sendMessage(setMode);
                                 }
                             }

--- a/addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/connector/RFXComSerialConnector.java
+++ b/addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/connector/RFXComSerialConnector.java
@@ -90,7 +90,8 @@ public class RFXComSerialConnector extends RFXComBaseConnector implements Serial
             readerThread.interrupt();
             try {
                 readerThread.join();
-            } catch (InterruptedException e) {}
+            } catch (InterruptedException e) {
+            }
         }
 
         if (out != null) {
@@ -117,6 +118,10 @@ public class RFXComSerialConnector extends RFXComBaseConnector implements Serial
 
     @Override
     public void sendMessage(byte[] data) throws IOException {
+        if (out == null) {
+            throw new IOException("Not connected sending messages is not possible");
+        }
+
         logger.trace("Send data (len={}): {}", data.length, DatatypeConverter.printHexBinary(data));
         out.write(data);
         out.flush();

